junit5/src/main/java/org/jboss/weld/junit5/auto/WeldJunit5AutoExtension.java

```diff
@@ -23,7 +23,30 @@
 
 import static org.jboss.weld.junit5.ExtensionContextUtils.getExplicitInjectionInfoFromStore;
 
-public class WeldJunit5AutoExtension extends org.jboss.weld.junit5.WeldJunit5Extension {
+import org.jboss.weld.junit5.WeldJunit5Extension;
+import org.jboss.weld.junit5.WeldJunitEnricher;
+
+/**
+ * An alternative to {@link WeldJunit5Extension} allowing to fully leverage annotation based configuration approach.
+ * When used, the extension will attempt to resolve all beans used in your test class and automatically adds them to
+ * Weld container while bootstrapping it.
+ * 
+ * There is quite a few annotations which can be used to configure it further still:
+ * @see ActivateScopes
+ * @see AddBeanClasses
+ * @see AddEnabledDecorators
+ * @see AddEnabledInterceptors
+ * @see AddExtensions
+ * @see AddPackages
+ * @see EnableAlternatives
+ * @see EnableAlternativeStereotypes
+ *
+ * Note that this approach cannot be combined with {@link WeldJunit5Extension}, choose one or the other approach, not both.
+ *
+ * @see EnableAutoWeld
+ * @see WeldJunitEnricher
+ */
+public class WeldJunit5AutoExtension extends WeldJunit5Extension {
```

Comment:
@kdubb I wonder - was it actually intended that you can now use `WeldJunitAutoExtension` along with stuff from `WeldJunit5Extension`?
E.g. that you can still define `@WeldSetup WeldInitiator` and configure deployment with that _and_ with annotations?

Reply:
Yes, it was intentional. I'm not sure it's required but I thought leaving a fallback for some edge case that annotations didn't cover was fine.

Reply:
Hmm, unless a use case surfaces, it may be safer to have abstract class and two different impls.
This possibly invites undefined behaviour or conflicting config of both.
Anyway, thanks for info.